modules/tensor_mechanics/doc/content/source/materials/crystal_plasticity/ComputeMultipleCrystalPlasticityStress.md

```diff
@@ -12,50 +12,91 @@ within a continuum mechanics framework enables the utilization of these models
+The Cauchy stress measure is then used in the traditional FEM residual calculation within the MOOSE
+framework (see [xtalpl_nr_pk2convergence]).
 
 A brief overview of the relevant continuum mechanics concepts for the crystal
-plasticity constitutive model base class is given before the crystal plasticity
-specific algorithm overview.
+plasticity constitutive model base class is given below.
+
+### Constitutive Equations
+
+Following [!cite](Asaro:1983kf) `ComputeMultipleCrystalPlasticityStress` uses a multiplicative
+decomposition of the deformation gradient into elastic and plastic components:
+\begin{equation}
+  \label{eqn:deformationGradDecomposition}
+  \mathbf{F} = \mathbf{F}^e \mathbf{F}^p
+\end{equation}
+The change in the crystal shape due to dislocation motion is accounted for in
+the plastic deformation gradient tensor, $\mathbf{F}^p$, while the elastic
+deformation gradient tensor, $\mathbf{F}^e$, accounts for recoverable elastic
+stretch and rotations of the crystal lattice. The evolution of the plastic
+deformation gradient is given as
+\begin{equation}
+  \label{eqn:plasticDeformationGradRate}
+  \dot{\mathbf{F}}^p = \mathbf{L}^p \mathbf{F}^p
+\end{equation}
+where $\mathbf{L}^p$ is the total plastic velocity gradient. The plastic deformation
+gradient rate is used to calculate the increment of the Lagrangian strain,
+[eqn:defLagrangianStrainRightCauchyGreen], for a given $\Delta t$ time step.
+
+
+The total plastic velocity gradient is computed as the sum of plastic velocity gradients coming from
+all the individual crystal plasticity constitutive equations,
+\begin{equation}
+\label{eqn:SumShears_multiple_cp}
+\mathbf{L^p} = \sum_{i=1}^m \mathcal{L^p_i},
+\end{equation}
+where $\mathcal{L^p_i}$ is the plastic velocity gradient that comes from one crystal plasticity constitutive equation,
+$m$ is the total number of crystal plasticity models. Here, $\mathcal{L^p_i}$  defined as the sum of the slip increments on
+all of the slip systems [!citep](Asaro:1983kf),
+\begin{equation}
+\label{eqn:SumShears_one_cp}
+\mathcal{L}^p_i = \sum_{\alpha=1}^n \dot{\gamma}_i^{\alpha} \mathbf{s}_{i,o}^{\alpha} \otimes \mathbf{m}_{i,o}^{\alpha}
+\end{equation}
+where $\dot{\gamma}^{\alpha}_i$ is the slip rate for the $i$ the model. Note that the associated slip direction and
+slip plane normal unit vectors, $\mathbf{s}_{i,o}$ and $\mathbf{m}_{i,o}$, are defined
+in the reference configuration. This relationship links the continuum framework
+to the constitutive models used in crystal plasticity.
 
+As a consequence of the decision to use the slip direction and plane normal unit
+vectors from the initial crystal orientation, [eqn:SumShears_one_cp], the second
+Piola-Kirchoff stress is used to determine the applied resolved shear stress,
+\begin{equation}
+  \label{eqn:appliedShearStress}
+  \tau^{\alpha}_i = \mathbf{S} : \mathbf{s}^{\alpha}_{i,o} \otimes \mathbf{m}^{\alpha}_{i,o}.
```

Comment:
I think the resolved shear stress should be `\tau = (\mathbf{F}^{e^T} \mathbf{P} \mathbf{F}^{p^T})  : \mathbf{s} \otimes \mathbf{m}`.

Reply:
Thanks @hugary1995 ! Just to be clear, may I ask what is the `\mathbf{P}` in your equation?

Reply:
Please check to make sure there is a consistent definition of the letter for the second PK stress through out the entire documentation if you make the change--whatever letter you select is fine with me so long as we just have one letter for all of the 2nd PK stress :-)

Reply:
I use P for PK1 and S for PK2.

Reply:
See below, I quickly wrote it down. I didn't check what the reference you cited there used, but that doesn't have to be ground truth -- we should do it correctly.
![image](https://user-images.githubusercontent.com/31555204/113037523-77a4a200-9163-11eb-8679-3a0c9de45473.png)

Reply:
@hugary1995 You made a very good point. I do not see any issue with your derivation, but the old implementation closely followed the formulation provided in Anand and Kalidindi 1992' paper. I am still trying to understand what causes the discrepancy, maybe based on a different assumption or something else. Let us together @hugary1995 @dewenyushu @sapitts look into this more closely. 

If we eventually agree to change the old formulation, I suggest we create another PR to fix it. In the meantime, we can create an issue to remind us any follow up work.

Reply:
That works for me. I'll create an issue for this.

Reply:
Thanks @hugary1995, for providing your derivation.

By looking closer to your derivation, I am wondering if you have an extra transpose in equation (2), i.e., should {Fp}^{-T} be (Fp)^{-1} instead? I am confused about equation (3), too. Should the right-hand-side be {Te}(Fp)^{-1}Fe instead? If I am right about the two equations, then in equation 6, the {Fe}^TP{Fp}^T term becomes PFeFp = PF = S, which is what we have in the current implementation.

Reply:
No problem Dewen, I added more details below. Let me know if there is anything wrong in the indicial notation:
![image](https://user-images.githubusercontent.com/31555204/113064904-ef83c400-9185-11eb-8bed-8e289d9edece.png)
Yea it also took me quite some time to pick up continuum mechanics!

Reply:
> If I am right about the two equations, then in equation 6, the {Fe}^TP{Fp}^T term becomes PFeFp = PF = S

No, PF != S, it should be FS = P if I remember correctly.

Reply:
I've got P = SF^T in Malvern, so the same as P=FS

Reply:
> > If I am right about the two equations, then in equation 6, the {Fe}^TP{Fp}^T term becomes PFeFp = PF = S
> 
> No, PF != S, it should be FS = P if I remember correctly.

More precisely, P = F^e S?

Reply:
Of course all the examples in my mechanics books are in the elasticity section, but I think we may use the total in the code...

@hugary1995 are you using T as the Cauchy stress in your example above? Double checking to make sure we're all on the same page

Reply:
> > > If I am right about the two equations, then in equation 6, the {Fe}^TP{Fp}^T term becomes PFeFp = PF = S
> > 
> > 
> > No, PF != S, it should be FS = P if I remember correctly.
> 
> More precisely, P = F^e S?

Hmmm, I don't think so. We always define P (PK1) as the conjugate of F, and S (PK2) as the conjugate of E (green lagrangian strain, in terms of F), regardless of decomposition of F, right? So it should always be P = FS I think.

Reply:
> @hugary1995 are you using T as the Cauchy stress in your example above? Double checking to make sure we're all on the same page

T^e and T^p are conjugates to F^e and F^p respectively, while Cauchy stress would be the conjugate to Euler-Almansi strain, right? If you really want a name for them, I think people usually relate T^p to the so called Mandel stress.

Reply:
Gotcha--thanks! I've always seen T as the Cauchy stress, so I'm glad you pointed out the definition (and now I see it in your write up too)

Reply:
> > If I am right about the two equations, then in equation 6, the {Fe}^TP{Fp}^T term becomes PFeFp = PF = S
> 
> No, PF != S, it should be FS = P if I remember correctly.

Oops, got it wrong from beginning to the end, lol... The derivation makes more sense to me now. Thanks!

Reply:
The "Gary" part looks correct. The "old" part seems to be different from what's in the markdown file though. Also should you use F instead of F^e to map Cauchy stress to PK2? Maybe we should find some time to discuss offline.

Reply:
> The "Gary" part looks correct. The "old" part seems to be different from what's in the markdown file though. Also shouldn't you use F instead of F^e to map Cauchy stress to PK2? Maybe we should find some time to discuss offline.

No, PK2 is defined on intermediate configuration, and it only pulls back the cauchy stress with Fe. Initially I though P will also be defined on intermediate configuration, but I realized that the PK1 in your formulation actually defines respect to the initial configuration.

Reply:
@jiangwen84 @hugary1995 please keep me in the loop. I would like to make sure that both the code and documentation are correct.

Reply:
So based on our discussion, when C^e is small, the two expressions are equivalent. This assumption seems to be fairly common in the crystal plasticity literature. The proof is essentially in @jiangwen84 's write-up. Thank you all for a nice discussion!